src/api/exchange/ExchangeApiMock.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import BN from 'bn.js'
+import assert from 'assert'
+
+import { DepositApiMock, BalancesByUserAndToken } from './DepositApiMock'
+import { ExchangeApi, Order, PlaceOrderParams, Erc20Api } from 'types'
+import { FEE_DENOMINATOR } from 'const'
+
+export interface OrdersByUser {
+  [userAddress: string]: Order[]
+}
+
+/**
+ * Basic implementation of Stable Coin Converter API
+ */
+export class ExchangeApiMock extends DepositApiMock implements ExchangeApi {
+  private registeredTokens: string[]
+  private tokenAddressToId: { [address in string]: number } // reverse mapping of registeredTokens for faster access
+  private maxTokens: number
+  private orders: OrdersByUser
+
+  public constructor(
+    balanceStates: BalancesByUserAndToken,
+    erc20Api: Erc20Api,
+    registeredTokens: string[] = [],
+    maxTokens: number = 10000,
+    ordersByUser: OrdersByUser = {},
+  ) {
+    super(balanceStates, erc20Api)
+    this.registeredTokens = registeredTokens
+    this.tokenAddressToId = registeredTokens.reduce((obj, address, index) => {
+      obj[address] = index
+      return obj
+    }, {})
+    this.maxTokens = maxTokens
+    this.orders = ordersByUser
+  }
+
+  public async getOrders(userAddress: string): Promise<Order[]> {
+    this._initOrders(userAddress)
+    return this.orders[userAddress]
+  }
+
+  public async getNumTokens(): Promise<number> {
+    return this.registeredTokens.length
+  }
+
+  /**
+   * Fee is 1/fee_denominator.
+   * i.e. 1/1000 = 0.1%
+   */
+  public async getFeeDenominator(): Promise<number> {
+    return FEE_DENOMINATOR
+  }
+
+  public async getTokenAddressById(tokenId: number): Promise<string> {
+    assert(typeof this.registeredTokens[tokenId] === 'string', 'Must have ID to get Address')
+    return this.registeredTokens[tokenId]
+  }
+
+  public async getTokenIdByAddress(tokenAddress: string): Promise<number> {
+    assert(typeof this.tokenAddressToId[tokenAddress] === 'number', 'Must have Address to get ID')
+    return this.tokenAddressToId[tokenAddress]
+  }
+
+  public async getCurrentPrice(_tokenId: number): Promise<BN> {
```

Comment:
Maybe we should remove this?
It's there an equivalent in the contract?

Reply:
I left it there in case we need it, because I couldn't find an easy way to implement this in the mock.
The contract has a mapping of tokens and prices which is updated when the solution is submitted:

```
/** @dev mapping of type tokenId -> curentPrice of tokenId */
    mapping (uint16 => uint128) public currentPrices;
```

I'll remove it for now. We can add it back if required.